source/solvers/vof.cc

```diff
@@ -0,0 +1,645 @@
+#include <core/bdf.h>
+#include <core/sdirk.h>
+#include <core/time_integration_utilities.h>
+#include <core/utilities.h>
+
+#include <solvers/vof.h>
+#include <solvers/vof_assemblers.h>
+#include <solvers/vof_scratch_data.h>
+
+#include <deal.II/base/work_stream.h>
+
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/mapping.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/solver_control.h>
+#include <deal.II/lac/sparsity_tools.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_solver.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+template <int dim>
+void
+VOF<dim>::assemble_matrix_and_rhs()
+{
+  assemble_system_matrix();
+  assemble_system_rhs();
+}
+
+
+template <int dim>
+void
+VOF<dim>::assemble_rhs()
+{
+  assemble_system_rhs();
+}
+
+template <int dim>
+void
+VOF<dim>::setup_assemblers()
+{
+  this->assemblers.clear();
+
+  // Time-stepping schemes
+  if (is_bdf(this->simulation_control->get_assembly_method()))
+    {
+      this->assemblers.push_back(
+        std::make_shared<VOFAssemblerBDF<dim>>(this->simulation_control));
+    }
+
```

Comment:
It would be good to throw an error here if an SDIRK type of scheme is used

Reply:
added.